### PR TITLE
Code refactoring for Spark Job remote debug with leverage Spark Job batch process

### PR DESCRIPTION
**Pull request by wezhang**

### CODE REVIEW

**Comment by djlan** [comment]
>                             isFailedJob = true;

Add break here?

---
Refers to: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/common/SparkSubmitHelper.java:150 in d37fede. [](commit_id = d37fede3972bb2f0e48b8ac2288bbad135349c51, deletion_comment = False)

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/run/SparkBatchJobDebuggerRunner.java [hunk 1]
```diff
@@ -23,92 +23,70 @@
...
+        }
+
+        boolean isDebugEnabled = Optional.of((RemoteDebugRunConfiguration) profile)
+                .map(RemoteDebugRunConfiguration::getSubmitModel)
+                .map(SparkSubmitModel::getAdvancedConfigModel)
+                .map(advModel -> advModel.enableRemoteDebug &&
+                        StringUtils.isNotEmpty(advModel.sshUserName) &&
+                        StringUtils.isNotEmpty(advModel.sshAuthType == SSHAuthType.UsePassword ?
+                                advModel.sshPassword :
+                                advModel.sshKeyFile.getPath()))
```
Comment (djlan) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/run/SparkBatchJobDebuggerRunner.java]: Make this a function of AdvancedConfigModel?

**Comment by wezhang** [comment]
to fix #957

File: Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/spark/run/SparkBatchJobRemoteDebugProcess.java [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+                // Debug session created and SSH port forwarded
+                getEventSubject().onNext(jdbReadyEvent);
+
+                return job;
+            } catch (JSchException e) {
+                // Rethrow it since JSch can't handle the certificate expired issue
+                throw new SparkJobException(
+                        "Can't create Spark Job remote debug session, " +
+                                "please check your SSH connect with manually login.",
+                        e);
```
Comment (djlan) [inline Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/spark/run/SparkBatchJobRemoteDebugProcess.java]: As discussed, can we check whether SSH password expired? Or at least add clear message, "Please check whether SSH password has expired, using Putty or other SSH tool"

**Comment by wezhang** [comment]
to fix #1546

**Pull request merged**